OpenZFS 2.3.0-rc3 & UCLIBC. Crash while compiling. MS_RDONLY = 1

<!-- Please fill out the following template, which will help other contributors address your issue. -->

<!--
Thank you for reporting an issue.

*IMPORTANT* - Please check our issue tracker before opening a new issue.
Additional valuable information can be found in the OpenZFS documentation
and mailing list archives.

Please fill in as much of the template as possible.
-->
Hi again,

I’m currently outside with very limited internet access and no computer, only a phone. Please excuse the brief explanation and lack of proper debugging.

The compilation crash for uClibc with OpenZFS 2.3.0-rc3 is as follows (note that it works correctly with glibc and musl):

```
In file included from tests/zfs-tests/cmd/getversion.c:22:
/tmp/tmp.s4lnjcKpqa/TestZfsUclibc/host/x86_64-buildroot-linux-uclibc/sysroot/usr/include/sys/mount.h:35:3: error: expected identifier before numeric constant
   35 |   MS_RDONLY = 1,                /* Mount read-only.  */
      |   ^~~~~~~~~
  CC       tests/zfs-tests/cmd/idmap_util.o
make[4]: *** [Makefile:8906: tests/zfs-tests/cmd/getversion.o] Error 1
make[4]: *** Waiting for unfinished jobs....                                  make[3]: *** [Makefile:12578: all-recursive] Error 1                          make[2]: *** [Makefile:4751: all] Error 2                                     make[1]: *** [package/pkg-generic.mk:289: /tmp/tmp.s4lnjcKpqa/TestZfsUclibc/build/zfs-2.3.0-rc3/.stamp_built] Error 2
make: *** [Makefile:23: _all] Error 2
E                                                                             ======================================================================
ERROR: test_run (tests.package.test_zfs.TestZfsUclibc.test_run)
----------------------------------------------------------------------        Traceback (most recent call last):                                              File "/home/jlsalvador/src/simplek8s-buildroot/buildroot/support/testing/infra/basetest.py", line 74, in setUp                                                self.b.build()                                                              File "/home/jlsalvador/src/simplek8s-buildroot/buildroot/support/testing/infra/builder.py", line 105, in build
    raise SystemError("Build failed")                                         SystemError: Build failed

----------------------------------------------------------------------        Ran 1 test in 471.419s                                                        
FAILED (errors=1)
```

### System information
<!--  add version after "|" character -->
Type | Version/Name
 --- | ---
Distribution Name	| buildroot
Distribution Version	| main
Kernel Version	| 5.15.35
Architecture	| x86-64
OpenZFS Version	| 2.3.0-rc3
<!--
Command to find OpenZFS version:
zfs version

Commands to find kernel version:
uname -r		# Linux
freebsd-version -r	# FreeBSD
-->

### Describe the problem you're observing

### Describe how to reproduce the problem

### Include any warning/errors/backtraces from the system logs
<!--
*IMPORTANT* - Please mark logs and text output from terminal commands
or else Github will not display them correctly.
An example is provided below.

Example:
```
this is an example how log text should be marked (wrap it with ```)
```
-->

--- a/tests/zfs-tests/cmd/getversion.c
+++ b/tests/zfs-tests/cmd/getversion.c
@@ -19,9 +19,13 @@
  */
 
 #include <sys/ioctl.h>
+#ifdef _KERNEL
+#include <sys/fcntl.h>
+#else
+#include <fcntl.h>
+#endif
 #include <linux/fs.h>
 #include <err.h>
-#include <fcntl.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>

